webapp/src/main/java/com/box/l10n/mojito/service/ai/openai/OpenAIPromptService.java

```diff
@@ -0,0 +1,169 @@
+package com.box.l10n.mojito.service.ai.openai;
+
+import com.box.l10n.mojito.JSR310Migration;
+import com.box.l10n.mojito.entity.AIPrompt;
+import com.box.l10n.mojito.entity.AIPromptContextMessage;
+import com.box.l10n.mojito.entity.AIPromptType;
+import com.box.l10n.mojito.entity.PromptType;
+import com.box.l10n.mojito.entity.Repository;
+import com.box.l10n.mojito.entity.RepositoryAIPrompt;
+import com.box.l10n.mojito.rest.ai.AIException;
+import com.box.l10n.mojito.rest.ai.AIPromptContextMessageCreateRequest;
+import com.box.l10n.mojito.rest.ai.AIPromptCreateRequest;
+import com.box.l10n.mojito.service.ai.PromptService;
+import com.box.l10n.mojito.service.repository.RepositoryRepository;
+import io.micrometer.core.annotation.Timed;
+import jakarta.transaction.Transactional;
+import java.time.ZonedDateTime;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Service;
+
+@Service
+@ConditionalOnProperty(value = "l10n.ai.service.type", havingValue = "OpenAI")
+public class OpenAIPromptService implements PromptService {
+
+  static Logger logger = LoggerFactory.getLogger(OpenAIPromptService.class);
+
+  @Autowired RepositoryRepository repositoryRepository;
+
+  @Autowired AIPromptRepository aiPromptRepository;
+
+  @Autowired AIPromptTypeRepository aiPromptTypeRepository;
+
+  @Autowired RepositoryAIPromptRepository repositoryAIPromptRepository;
+
+  @Autowired AIPromptContextMessageRepository aiPromptContextMessageRepository;
+
+  @Timed("OpenAIPromptService.createPrompt")
+  @Transactional
+  public Long createPrompt(AIPromptCreateRequest AIPromptCreateRequest) {
+
+    Repository repository =
+        repositoryRepository.findByName(AIPromptCreateRequest.getRepositoryName());
+
+    if (repository == null) {
+      logger.error("Repository not found: {}", AIPromptCreateRequest.getRepositoryName());
+      throw new AIException("Repository not found: " + AIPromptCreateRequest.getRepositoryName());
+    }
+
+    AIPromptType aiPromptType =
+        aiPromptTypeRepository.findByName(AIPromptCreateRequest.getPromptType());
+    if (aiPromptType == null) {
+      logger.error("Prompt type not found: {}", AIPromptCreateRequest.getPromptType());
+      throw new AIException("Prompt type not found: " + AIPromptCreateRequest.getPromptType());
+    }
+
+    AIPrompt aiPrompt = new AIPrompt();
+    aiPrompt.setSystemPrompt(AIPromptCreateRequest.getSystemPrompt());
+    aiPrompt.setUserPrompt(AIPromptCreateRequest.getUserPrompt());
+    aiPrompt.setPromptTemperature(AIPromptCreateRequest.getPromptTemperature());
+    aiPrompt.setModelName(AIPromptCreateRequest.getModelName());
+    ZonedDateTime now = JSR310Migration.dateTimeNow();
```

Comment:
Technically not needed - as the table as you've defined it would provide a value for the field.